pkg/restapi/vc/operation/operations.go

```diff
@@ -322,29 +346,65 @@ func (o *Operation) storeVCHandler(rw http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	doc := operation.StructuredDocument{}
-	doc.Content = make(map[string]interface{})
-	doc.Content["message"] = data.Credential
+	doc, err := o.buildStructuredDoc(data, vc)
+	if err != nil {
+		o.writeErrorResponse(rw, http.StatusBadRequest, err.Error())
 
-	// We need an EDV-compliant ID. The UUID that's in the VC is a good fit, since it's 128 bits long already.
-	// Let's convert it to base58 and use it as the StructuredDocument ID.
-	edvDocID, err := convertVCIDToEDVCompatibleFormat(vc.ID)
+		return
+	}
+
+	marshalledStructuredDoc, err := json.Marshal(doc)
 	if err != nil {
 		o.writeErrorResponse(rw, http.StatusBadRequest, err.Error())
 
 		return
 	}
 
-	doc.ID = edvDocID
+	// We have no recipients, so we pass in the sender key as the recipient key as well
+	encryptedStructuredDoc, err := o.packer.Pack(marshalledStructuredDoc,
+		base58.Decode(o.senderKey), [][]byte{base58.Decode(o.senderKey)})
+	if err != nil {
+		o.writeErrorResponse(rw, http.StatusBadRequest, err.Error())
 
-	_, err = o.client.CreateDocument(data.Profile, &doc)
+		return
+	}
+
+	encryptedDocument := operation.EncryptedDocument{
+		ID:       doc.ID,
+		Sequence: 0,
+		JWE:      encryptedStructuredDoc,
+	}
+
+	_, err = o.client.CreateDocument(data.Profile, &encryptedDocument)
 	if err != nil {
 		o.writeErrorResponse(rw, http.StatusBadRequest, err.Error())
 
 		return
 	}
 }
 
+func (o *Operation) buildStructuredDoc(data *StoreVCRequest,
+	vc *verifiable.Credential) (*operation.StructuredDocument, error) {
+	// We need an EDV-compliant ID. The UUID that's in the VC is a good fit, since it's 128 bits long already.
+	// Let's convert it to base58 and use it as the StructuredDocument ID.
+	edvDocID, err := convertVCIDToEDVCompatibleFormat(vc.ID)
```

Comment:
I’m concerned about this. Is this identifier stored encrypted or unencrypted?

Reply:
This identifier is not encrypted. Let me know your thoughts on this.

Reply:
Cannot use the credential ID in that case.

Reply:
@sandrask proposed a potential solution. We can hash the VC id and use that instead. This way, you can't derive the VC ID from the EDV document ID. However, if you know the VC ID then you can still retrieve the document. And this way I don't have to store a mapping of VC IDs to EDV document IDs somewhere. 

I've updated the PR with hashing. Let me know what you think. @troyronda @sandrask @Baha-sk

Reply:
(scroll down to the convertVCIDToEDVCompatibleFormat function)